docs-generation/scripts/update_param_translations.py

```
@@ -139,11 +139,13 @@ def update_doc_with_translations(doc_file, node_name, lang, frontend_translation
             lines = content.split('\n')
             in_input_section = False
             for i, line in enumerate(lines):
-                # Detect if we're in the Inputs section
-                if re.match(r'##\s+(?:输入|輸入|入力|입력|Входы|Entradas|Entrées|Inputs|المدخلات|Girdiler|ورودی‌ها)', line):
+                # Detect if we're in the Inputs section.
+                # Only H2 (## X) opens/closes a section; H3 (### X) subheadings
+                # like "### Common Inputs" must NOT close it.
+                if re.match(r'^##\s+(?:输入|輸入|入力|입력|Входы|Entradas|Entrées|Inputs|المدخلات|Girdiler|ورودی‌ها)', line):
```

Ruff (0.16.1)

[warning] String contains ambiguous `ه` (ARABIC LETTER HEH). Did you mean `o` (LATIN SMALL LETTER O)?

(RUF001)